diff --git a/.dockerignore b/.dockerignore
index d2e8d95..e5c32cc 100644
--- a/.dockerignore
+++ b/.dockerignore
@@ -8,3 +8,4 @@
 !/LICENSE
 !/NOTICE.md
 !/README.md
+!qemu-aarch64-static-3.0.0
diff --git a/Dockerfile b/Dockerfile
index e54a2a1..3e28c3a 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,7 @@
 # stage: builder
-FROM golang:alpine AS builder
+FROM resin/aarch64-alpine-golang AS builder
+COPY ./qemu-aarch64-static-3.0.0 /usr/bin/qemu-aarch64-static
+RUN [ "cross-build-start" ]
 
 COPY . /go/src/github.com/concourse/docker-image-resource
 ENV CGO_ENABLED 0
@@ -11,9 +13,12 @@ RUN set -e; \
     for pkg in $(go list ./...); do \
       go test -o "/tests/$(basename $pkg).test" -c $pkg; \
     done
+RUN [ "cross-build-end" ]
 
 # stage: resource
-FROM alpine:edge AS resource
+FROM resin/aarch64-alpine:edge AS resource
+COPY ./qemu-aarch64-static-3.0.0 /usr/bin/qemu-aarch64-static
+RUN [ "cross-build-start" ]
 RUN apk --no-cache add \
       bash \
       docker \
@@ -23,15 +28,18 @@ RUN apk --no-cache add \
     ;
 COPY --from=builder /assets /opt/resource
 RUN ln -s /opt/resource/ecr-login /usr/local/bin/docker-credential-ecr-login
+RUN [ "cross-build-end" ]
 
 # stage: tests
 FROM resource AS tests
+RUN [ "cross-build-start" ]
 COPY --from=builder /tests /tests
 ADD . /docker-image-resource
 RUN set -e; \
     for test in /tests/*.test; do \
       $test -ginkgo.v; \
     done
+RUN [ "cross-build-end" ]
 
 # final output stage
 FROM resource
