diff --git a/Dockerfile b/Dockerfile
index 5f1ca42..e5c81c6 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,6 @@
-FROM alpine:edge AS tunnelbuilder
+FROM resin/aarch64-alpine:3.8 AS tunnelbuilder
+COPY ./qemu-aarch64-static-3.0.0 /usr/bin/qemu-aarch64-static
+RUN [ "cross-build-start" ]
 RUN apk --no-cache add git make gcc g++ libressl-dev
 
 WORKDIR /root
@@ -6,8 +8,11 @@ RUN git clone https://github.com/proxytunnel/proxytunnel.git
 
 WORKDIR /root/proxytunnel
 RUN make
+RUN [ "cross-build-end" ]
 
-FROM alpine:edge AS resource
+FROM resin/aarch64-alpine:edge AS resource
+COPY ./qemu-aarch64-static-3.0.0 /usr/bin/qemu-aarch64-static
+RUN [ "cross-build-start" ]
 
 RUN apk --no-cache add \
   bash \
@@ -194,15 +199,6 @@ WORKDIR         /usr/lib
 RUN             rm -rf \
                     perl \
                     perl5
-
-FROM resource AS tests
-ADD test/ /tests
-RUN /tests/all.sh
-
-FROM resource AS integrationtests
-RUN apk --no-cache add squid
-ADD test/ /tests/test
-ADD integration-tests /tests/integration-tests
-RUN /tests/integration-tests/integration.sh
+RUN [ "cross-build-end" ]
 
 FROM resource
